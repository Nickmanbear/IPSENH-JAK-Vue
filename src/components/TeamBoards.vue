<template>
  <div class="teamBoards">
    <br>
    <h1>{{ team.name }}</h1>
    <BoardPreview
      v-for="board in boards" :key="board.id"
      v-bind:board="board"
      @deleted="removeBoardPreview()"/>
  </div>
</template>

<script>
// @ is an alias to /src
import BoardPreview from '@/components/BoardPreview.vue';
import axios from '@/axiosInstance';

export default {
  name: 'TeamBoards',
  components: {
    BoardPreview,
  },
  props: {
    team: {},
  },
  data() {
    return {
      boards: [],
    };
  },
  mounted() {
    this.getBoards();
  },
  methods: {
    getBoards() {
      axios.get(`/board/team/${this.team.id}`)
        .then((response) => {
          this.boards = response.data;
        });
    },
    removeBoardPreview() {
      this.getBoards();
    },
  },
};
</script>
