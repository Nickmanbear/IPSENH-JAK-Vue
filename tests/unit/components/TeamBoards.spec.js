import { shallowMount } from '@vue/test-utils';
import TeamBoards from '@/components/TeamBoards.vue';

jest.mock('axios', () => ({
  create: () => ({
    get: () => Promise.resolve({
      data: [
        { boardId: 1, name: 'Board 1' },
        { boardId: 2, name: 'Board 2' },
        { boardId: 3, name: 'Board 3' },
      ],
    }),
  }),
}));

describe('TeamBoards', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(TeamBoards, {
      propsData: {
        team: {
          id: 1,
          name: 'test team',
          leader: {
            id: 1,
            username: 'teamleader',
          },
          members: [
            {
              id: 1,
              username: 'teamleader',
            },
            {
              id: 2,
              username: 'member1',
            },
            {
              id: 3,
              username: 'member2',
            },
          ],
        },
      },
      data() {
        return {
          boards: [],
        };
      },
    });
  });

  it('should get boards', async () => {
    await cmp.vm.getBoards();
    expect(cmp.vm.boards).toEqual([
      { boardId: 1, name: 'Board 1' },
      { boardId: 2, name: 'Board 2' },
      { boardId: 3, name: 'Board 3' },
    ]);
  });

  it('should delete board', async () => {
    await cmp.vm.removeBoardPreview();
    expect(cmp.vm.boards).toEqual([
      { boardId: 1, name: 'Board 1' },
      { boardId: 2, name: 'Board 2' },
      { boardId: 3, name: 'Board 3' },
    ]);
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
