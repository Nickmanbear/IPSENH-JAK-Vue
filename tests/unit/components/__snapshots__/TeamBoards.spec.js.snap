// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamBoards has the expected html structure 1`] = `
<div
  class="teamBoards"
>
  <br />
   
  <h1>
    test team
  </h1>
   
  <boardpreview-stub
    board="[object Object]"
  />
  <boardpreview-stub
    board="[object Object]"
  />
  <boardpreview-stub
    board="[object Object]"
  />
</div>
`;
